test/Datadog.Trace.ClrProfiler.IntegrationTests/HttpClientTests.cs

```diff
@@ -0,0 +1,66 @@
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using Datadog.Trace.TestHelpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests
+{
+    public class HttpClientTests : TestHelper
+    {
+        public HttpClientTests(ITestOutputHelper output)
+            : base("HttpMessageHandler", output)
+        {
+        }
+
+        [Fact]
+        [Trait("Category", "EndToEnd")]
+        public void SubmitsTracesWithHttpClient()
+        {
+            const int agentPort = 9006;
+
+            using (var agent = new MockTracerAgent(agentPort))
+            using (ProcessResult processResult = RunSampleAndWaitForExit(agentPort, "HttpClient"))
```

Comment:
Just throwing the idea, I'm wondering if it would make sense to give samples the ability to return serialized data on stdout (or something else) that we could assert on. For instance, here we could have the sample write all the headers of the http requests received by `HandleHttpRequests` and we'd be able to check if the context propagation is behaving properly.

Reply:
The sample app already [outputs the headers to stdout](https://github.com/DataDog/dd-trace-dotnet/pull/267/files/01f3c9080da484a39778ed3fd975d1a84c1aeb40#diff-e6aa75f5d4339434ec335af46fa76edaR69) (although in a more human-readable format), and `ProcessResult` includes the entire contents of stdout, so this should be trivial to implement.

Reply:
Added asserts on the http headers: https://github.com/DataDog/dd-trace-dotnet/pull/267/commits/515e4a6e90a41ed6897062e83699f0d8a7433b6c